<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SkeletonGraspPlanner</class>
 <widget class="QMainWindow" name="SkeletonGraspPlanner">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1079</width>
    <height>819</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Skeleton Grasp Planner</string>
  </property>
  <property name="toolTip">
   <string>Plan grasps for a batch of models</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QFrame" name="frameViewer">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QFrame" name="frame_2">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>120</height>
          </size>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>Model</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="pushButtonLoadData">
               <property name="text">
                <string>load</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonReset">
               <property name="text">
                <string>reset</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonSave">
               <property name="text">
                <string>save</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>End Effector</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>2</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="pushButtonOpen">
               <property name="text">
                <string>open</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonClose">
               <property name="text">
                <string>close</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QToolBox" name="toolBox">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>200</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="options">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>276</width>
            <height>206</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Grasp Planning Setup</string>
          </attribute>
          <widget class="QFrame" name="frameGrasping">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>4</y>
             <width>274</width>
             <height>141</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <property name="spacing">
             <number>2</number>
            </property>
            <item row="1" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Nr of Grasps</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QSpinBox" name="spinBoxTimeOut">
              <property name="minimum">
               <number>0</number>
              </property>
              <property name="maximum">
               <number>10000000</number>
              </property>
              <property name="value">
               <number>30</number>
              </property>
             </widget>
            </item>
            <item row="4" column="0" colspan="2">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Timeout, seconds (0=endless)</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QDoubleSpinBox" name="doubleSpinBoxQuality">
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="maximum">
               <double>1.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.001000000000000</double>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QCheckBox" name="checkBoxFoceClosure">
              <property name="text">
               <string>force closure</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QSpinBox" name="spinBoxGraspNumber">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10000000</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Min quality</string>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QCheckBox" name="checkBoxEvaluateGrasps">
              <property name="text">
               <string>evaluate grasps</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="QCheckBox" name="checkBoxVerbose">
              <property name="text">
               <string>verbose</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="visu">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>276</width>
            <height>206</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Visualization</string>
          </attribute>
          <widget class="QFrame" name="frameVisualization">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>285</width>
             <height>141</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBoxColModel">
              <property name="text">
               <string>collision model</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QCheckBox" name="checkBoxHand">
              <property name="text">
               <string>show hand</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBoxCones">
              <property name="text">
               <string>friction cones</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QCheckBox" name="checkBoxGrasps">
              <property name="text">
               <string>show grasps</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBoxGraspingInterval">
              <property name="text">
               <string>grasping interval</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QCheckBox" name="checkBoxGCP">
              <property name="text">
               <string>show GCP</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="3">
             <widget class="QCheckBox" name="checkBoxGraspingPlane">
              <property name="text">
               <string>grasping plane + projected points</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>transparency</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1" colspan="2">
             <widget class="QSlider" name="horizontalSliderTr">
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>30</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
         <widget class="QWidget" name="skeleton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>276</width>
            <height>206</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Skeleton</string>
          </attribute>
          <widget class="QFrame" name="frameSkeleton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>0</y>
             <width>261</width>
             <height>101</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>100</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QRadioButton" name="radioButtonSkeleton">
              <property name="text">
               <string>show skeleton</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButtonSegmentation">
              <property name="text">
               <string>show segmentation</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButtonNothing">
              <property name="text">
               <string>don't show</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Grasp Planning</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushButtonOptions">
            <property name="text">
             <string>options</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="widget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QPushButton" name="pushButtonPlan">
               <property name="maximumSize">
                <size>
                 <width>70</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>plan</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonPlanAll">
               <property name="toolTip">
                <string/>
               </property>
               <property name="text">
                <string>plan all</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButtonPlanBatch">
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string>plan batch</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
         <zorder>widget</zorder>
         <zorder>pushButtonOptions</zorder>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="widget_2" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>current grasp</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBoxGraspNumberPlanned">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>0</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelInfo">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>60</height>
          </size>
         </property>
         <property name="text">
          <string>Grasps: 0</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="labelSkeleton">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>60</height>
          </size>
         </property>
         <property name="text">
          <string>Skeleton Info</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1079</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
